### PR TITLE
color-scheme-generator: init at 0.9.2.1

### PR DESCRIPTION
**Pull request by nikolaizombie1**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/co/color-scheme-generator/package.nix [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    homepage = "https://github.com/nikolaizombie1/color_scheme_generator";
+    license = licenses.gpl3;
+    maintainers = [
+      {
+        name = "Fabio J. Matos Nieves";
+        email = "fabio.matos999@gmail.com";
+        githubId = "70602908";
+        github = "nikolaizombie1";
+      }
+    ];
```
Comment (getchoo) [inline pkgs/by-name/co/color-scheme-generator/package.nix:42]:
```suggestion
    maintainers = with lib.maintainers; [ nikolaizombie1 ];
```
We don't really inline things like this. You should instead make a separate commit adding yourself to [maintainer-list.nix](https://github.com/NixOS/nixpkgs/blob/master/maintainers/maintainer-list.nix), per our [contributing guide](https://github.com/NixOS/nixpkgs/blob/239a55c50ac48b22175722b62a7f79e211cf5fd3/CONTRIBUTING.md#commit-conventions)

File: pkgs/by-name/co/color-scheme-generator/package.nix [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    repo = "color_scheme_generator";
+    rev = "v${version}";
+    hash = "sha256-tSoM6MP58vOoKRMQjNq8nKMiVTfS9N/mckchAPcUM0g=";
+  };
+
+  cargoHash = "sha256-A8ld2c8ZcqaZN29Fz085FQy8UpugIDMOKT01r+SfCcc=";
+
+  nativeBuildInputs = [ perl ];
+
+  meta = with lib; {
```
Comment (getchoo) [inline pkgs/by-name/co/color-scheme-generator/package.nix:23]:
```suggestion
  meta = {
```
Nitpick, but we try to avoid larger `with ...;` scopes like this

File: pkgs/by-name/co/color-scheme-generator/package.nix [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "color-scheme-generator";
+  version = "0.9.2.1";
+
+  src = fetchFromGitHub {
+    owner = "nikolaizombie1";
+    repo = "color_scheme_generator";
+    rev = "v${version}";
```
Comment (getchoo) [inline pkgs/by-name/co/color-scheme-generator/package.nix:15]:
```suggestion
    tag = "v${version}";
```
This is a new argument for `fetchFromGitHub` that we should use instead when referring to tags (as to avoid branches of the same name, for example)

**Review by getchoo — CHANGES_REQUESTED** [review]
Other than that, LGTM and welcome to Nixpkgs!

**Comment by nikolaizombie1** [comment]
@getchoo  Addressed the concerns.

**Comment by getchoo** [comment]
The commit message should also be `color-scheme-generator: init at 0.9.2.1` (without the v prefix)

**Comment by nikolaizombie1** [comment]
@getchoo Addressed the concerns.

**Review by getchoo — COMMENTED** [review]
Hi @nikolaizombie1, the package itself LGTM right now, but it seems there are some issues with the git history. Could you squash this down to the maintainer list commit and the addition of the package? There are also some extra changes to maintainer's list that I'm not sure need to be there

I would recommend rebasing onto the latest `master` branch before squashing, like so

```console
$ git fetch origin master
$ git rebase -i origin/master
```

And feel free to reach out if you have any trouble with this :)

**Comment by nikolaizombie1** [comment]
@getchoo I think I was able to rebase main into my branch. I think.
